luxonis_ml/data/datasets/annotation.py

```diff
@@ -92,6 +93,39 @@ def combine_to_numpy(
         pass
 
 
+class TextAnnotation(Annotation):
+    _label_type = LabelType.TEXT
+    type_: Literal["text"] = Field("text", alias="type")
+
+    text: str
+    max_len: int
+
+    def to_numpy(
+        self,
+        class_mapping: Dict[str, int],
+        **_,
+    ) -> np.ndarray:
+        text_label = np.zeros(self.max_len)
+        for idx, char in enumerate(self.text):
+            class_ = class_mapping.get(char, 0)
+            text_label[idx] = class_
+        return text_label
+
+    @staticmethod
+    def combine_to_numpy(
+        annotations: List["TextAnnotation"],
+        class_mapping: Dict[str, int],
+        **_,
+    ) -> np.ndarray:
+        text_labels = None
```

Comment:
`text_labels` can be already set to `np.zeros((len(annotations), ann.max_len))` so there's no chance to return `None`